addon/components/frost-object-browser.js

```diff
@@ -45,11 +48,21 @@ export default Component.extend({
   // == Actions ===============================================================
   actions: {
     collapseFilter () {
-      this.onFilterVisibleChange(false)
+      this.set('isFilterHiddenOnLoad', true)
+
+      const callback = this.onFilterHide
```

Comment:
Prefer

```
if (this.onFilterHide) {
  this.onFilterHide()
}
```

Reply:
why one more look up?

Reply:
It does not increase the lookup count anymore than what you had.  If we were calling `this.get('onFilterHide')` you could make the argument that looking it up once and retaining the value is beneficial but when accessing `this.onFilterHide` directly it is exactly that - direct.   So in your code we are setting the value of `this.onFilterHide` to another variable so that we can then check its existence and then call it.  That is what my code is doing without the use of an unnecessary assignment.

Reply:
I (now more fully) understand what you are saying.  That aside, and now perhaps moving into a more academic discussion, what is the difference in memory usage between the assignment of a property to another that is then invoked vs the lookup off of `this` for invocation?  If there is a difference we should pick the better one.  If we do not know my proposal appears visually cleaner to me.

Reply:
I'm following the basic convention when there's need for more than one javascript heap lookup, I'll assign it to a variable. Otherwise, I'll use it directly.

per discussion.
1. what is the difference in memory usage
Assign the value to a new variable, of course, takes more memory. But all it takes is the space for one variable, and this memory gets freed up as soon as js engine quits the callback function.

2. which one is faster
The result is very close when comparing one variable assign vs one property look up. Variable assigning is slightly faster.
![screen shot 2017-10-25 at 4 30 04 pm](https://user-images.githubusercontent.com/14845784/32022729-432d0b28-b9a5-11e7-8f54-b16942e75ed7.png)

The difference becomes more obvious when there is more than 2 lookup.
![screen shot 2017-10-25 at 4 43 40 pm](https://user-images.githubusercontent.com/14845784/32022779-69c3878a-b9a5-11e7-9acd-88ace5793a58.png)

Reply:
Your original implementation is the clear winner and should remain :-)